File: C#/TddApi/NavigationApp/s14.refactor_unit_tests/Navigator.cs
using System;

namespace NavigationApp.s14.refactor_unit_tests
{
    public class Navigator
    {
        DistanceProvider distProvider;
        Location startingPoint;

        public Navigator(Location loc, DistanceProvider distService)
        {
            distProvider = distService;
            startingPoint = loc;
        }

        public Distance getDistanceFromDestination()
        {
            return distProvider.getDistance(startingPoint, startingPoint);
        }

        public void setDestination(Location dest)
        {
            
        }
    }
}

File: Python/s09_refactor_first_test/navigator.py
from s09_refactor_first_test.distance import Distance


class Navigator:
    def __init__(self, loc, dist_provider):
        self.dist_provider = dist_provider
        pass

    def getDistanceFromDestination(self):
        return Distance(0)

    def setDestination(self, dest):
        pass
